core/CMakeLists.txt

```diff
@@ -1,29 +1,25 @@
 # core/CMakeLists.txt
 # Cubos core build configuration
 
-option(WITH_GLFW "With GLFW?" ON)
-option(WITH_OPENGL "With OpenGL?" ON)
+# ----------------------- Declare core library options ------------------------
 
-if(WITH_GLFW)
-	option(GLFW_USE_SUBMODULE "Compile GLFW from source?" ON)
-endif()
-
-option(GLM_USE_SUBMODULE "Compile GLM from source?" ON)
+option(CUBOS_CORE_SAMPLES "Build cubos::core samples" ${PROJECT_IS_TOP_LEVEL})
+option(CUBOS_CORE_TESTS "Build cubos::core tests?" ${PROJECT_IS_TOP_LEVEL})
+option(CUBOS_CORE_SHARED "Build cubos::core as shared library?" ON)
 
-set(CUBOS_CORE_ECS_MAX_COMPONENTS "63" CACHE STRING "The maximum number of components registered in an ECS world.")
+option(CUBOS_CORE_GLFW "Build cubos::core with GLFW support?" ON)
+option(CUBOS_CORE_OPENGL "Build cubos::core with OpenGL support?" ON)
```

Comment:
This is more of a question than a comment: why do we provide an option for some very important dependencies? Wouldn't not building these affect the core features of CUBOS?

Reply:
Yes, it would mean you're building without graphics/window support. In the future, we might provide alternatives to OpenGL for example - `CUBOS_CORE_VULKAN`, `CUBOS_CORE_DIRECTX12` etc. When that happens this will make more sense